### PR TITLE
TEAMROCKET#15+#16: Firebase Auth + CRUD for Firebase 

### PR DESCRIPTION
**Pull request by DanlandYZ**
modifies getAllRecipes (probably need further modifications) 
user can add recipes to their favorites, and it will sync with database even in the case of accidentally close the window.

### CODE REVIEW

**Comment by DanlandYZ** [comment]
updated most JavaScript files replacing any local storage functions with calls to database. 
Still have linting errors need to be fixed during weekends, 
will double check js files during weekends too.

**Review by gmihir — CHANGES_REQUESTED** [review]
This is awesome! I found one issue though: when a user favorites a recipe (logged in) and then navigates to the my account page, the favorited recipe does not show up.

**Comment by gmihir** [comment]
Additionally, I am having trouble creating a recipe.

Uncaught (in promise) TypeError: Cannot set properties of null (setting '0a2d5a7c-8281-4b0c-9719-7bd85da8861c')
    at createRecipe (utils.js:370:25)
    at async HTMLButtonElement.<anonymous> (createRecipe.js:139:24)

**Comment by DanlandYZ** [comment]
Pretty much all done.

**Review by gmihir — APPROVED** [review]
this is good to go! of course we will need to add some unit tests before release

File: testing/unit/utilsTesting.test.js [hunk 1]
```diff
@@ -28,12 +28,6 @@ test('assert getAllRecipes() succeeds', async () => {
     expect(recipes.length).toBe(3);
 });
 
-test('assert readRecipe() returns correct recipe', async () => {
```
Comment (lramos0) [inline testing/unit/utilsTesting.test.js:31]: Unit tests broken as a result of implementing CRUD related operations for firebase, expected and known. To be fixed before release.

**Review by lramos0 — APPROVED** [review]
Approving for the sake of unblocking personalization related features. We still have a insufficient unit test suite.

**Comment by lramos0** [comment]
I hate testing... but here ya go, have fun!

**Comment by DanlandYZ** [comment]
this is really wired, the test run locally and test run github are failing different tests
Tests running on github are giving 2, 3, or 4 number of failing tests. 
not really sure what is going on.

**Review by ahthai2468 — APPROVED** [review]
I believe e2e is in a solid position for merging now - of course, we will need new unit testing.

**Comment by gmihir** [comment]
> I believe e2e is in a solid position for merging now - of course, we will need new unit testing.

ok thanks! i want to merge everything else in first bc this will probably cause a few conflicts

**Comment by ahthai2468** [comment]
> > I believe e2e is in a solid position for merging now - of course, we will need new unit testing.
> 
> ok thanks! i want to merge everything else in first bc this will probably cause a few conflicts

Sounds good, I am a bit concerned about unit testing for this. Are we doing unit testing after merging this or before? Will we have the bandwidth to have any semblance of unit testing for this lol

**Pull request merged**